runtime/lua/vim/lsp/client.lua

```diff
@@ -478,6 +476,21 @@ function Client.create(config)
     messages = { name = name, messages = {}, progress = {}, status = {} },
   }
 
+  --- @deprecated remove in 0.12
+  self.dynamic_capabilities = {
```

Comment:
I think if we want to remove this in 0.12 we need at least a replacement for `get`.  What fzf-lua, actions-preview and lspsaga are using this for looks quite reasonable to me and there isn't really an alternative.

So either only deprecate the inner properties except the `get` function, or we could drop the `_` prefix from `_get_dynamic_capability` and add docs for it.

Reply:
Based on:

```lua
  local supports_resolve = utils.__HAS_NVIM_010
      -- runtime/lua/lsp/buf.lua:on_user_choice
      and (function()
        ---@var choice {action: lsp.Command|lsp.CodeAction, ctx: lsp.HandlerContext}
        local ms = require("vim.lsp.protocol").Methods
        local choice = self.opts._items[idx]
        local bufnr = assert(choice.ctx.bufnr, "Must have buffer number")
        local reg = client.dynamic_capabilities:get(ms.textDocument_codeAction, { bufnr = bufnr })
        return utils.tbl_get(reg or {}, "registerOptions", "resolveProvider")
            or client.supports_method(ms.codeAction_resolve)
      end)()
      -- prior to nvim 0.10 we could check `client.server_capabilities`
      or utils.tbl_get(client.server_capabilities, "codeActionProvider", "resolveProvider")
```

and 

```lua
function act:support_resolve(client)
  if vim.version().minor >= 10 then
    local reg = client.dynamic_capabilities:get('textDocument/codeAction', { bufnr = ctx.bufnr })
    return vim.tbl_get(reg or {}, 'registerOptions', 'resolveProvider')
      or client.supports_method('codeAction/resolve')
  end
  return vim.tbl_get(client.server_capabilities, 'codeActionProvider', 'resolveProvider')
end
```

To me look the same as:

```lua
client:_supports_method(ms.textDocument_codeAction, { bufnr = bufnr })
```

Reply:
Ok, I've made a change to `client.support_method` so it work properly for resolve requests. WDYT?

Reply:
Add `Client:get_method_registration_options`.